### ISSUE TITLE
Github rate limit when checking for updates

### ISSUE THREAD

**Issue by adrianmarinica** [intercom/intercom-cordova#245]
## Version info
  - Intercom for Cordova/PhoneGap version: 4.1.2 (should reproduce with 5.0.1 as well)

## Expected behavior
Should correctly build the app, including the bundle for string translations and images.

## Actual behavior
Doesn't always include the bundle for string translations and images.

## Steps to reproduce
We are using a CI system called Bitrise (bitrise.io) for building our Cordova app. In the Intercom Cordova plugin, there is a [hook](https://github.com/intercom/intercom-cordova/blob/master/intercom-plugin/scripts/checkForUpdate.js) that runs before each build. This hook checks for updates and installs them, if there are any. The code section that handles this is highlighted below:

```
fetchLatestRelease(context, function(releaseData) {
        updateInfo.lastCheckDate = Date.now();

        if (releaseData != null) {
          updateInfo.releaseDate = Date.parse(releaseData['published_at']);
          updateIntercomIfNeeded(context, updateInfo, function() {
            deferral.resolve();
          });
        } else {
          writeUpdateInfo(context, updateInfo, function() {
            deferral.resolve();
          });
        }
});
```

The `fetchLatestRelease` call makes an HTTP request to Github to retrieve the version information, at `api.github.com/repos/intercom/intercom-ios/releases/latest`. The problem is that apparently Github has a rate limit for requests to its API. This rate is probably never hit when running the build locally, but since we're running the build through a CI system used by thousands of projects worldwide, then that rate limit is almost always hit. The Github response is basically this:

```
     status: '403 Forbidden',
     'x-ratelimit-limit': '60',
     'x-ratelimit-remaining': '0',
     'x-ratelimit-reset': '1518028172',
```
Receiving a non-200 response, the promise from the hook resolves in a null, which makes the hook not to call the `updateIntercomIfNeeded` method. This in turn makes the strings and images bundle not to be part of the app itself.

Is there a workaround for this behaviour? Or would you accept a PR for updating Intercom even if the status of that request is not 200?

**Comment by ronocod** [comment]
I don't think there's a workaround for this right now, a PR to fix this would be great :+1:

**Comment by adrianmarinica** [comment]
@ronocod Glad to help, I created #246 :)

**Comment by ronocod** [comment]
This is released as part of 5.0.2, can you make sure you don't experience this problem on that version?

**Comment by adrianmarinica** [comment]
@ronocod Yes, this release indeed fixed the issue I was describing above. Sadly, it introduced another issue, because I had to upgrade android-cordova to 7.0.0. I will investigate that problem some more and if I find something is amiss with the intercom-cordova plugin, I will open another GitHub issue. Thanks for the help! :)
